boot/bootutil/src/loader.c

```diff
@@ -43,7 +43,7 @@
 
 static struct boot_loader_state boot_data;
 
-#if defined(MCUBOOT_VALIDATE_SLOT0) && !defined(MCUBOOT_OVERWRITE_ONLY)
+#if defined(CONFIG_MCUBOOT_VALIDATE_SLOT0) && !defined(MCUBOOT_OVERWRITE_ONLY)
```

Comment:
The problem with this approach is that bootutil is the common library that implements mcuboot functionality, and cannot contain any Zephyr-specific code. (The Zephyr bootloader app is in boot/zephyr/, and you've already seen the Zephyr samples/zephyr).